grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -1025,6 +1025,21 @@ class ProductAvailabilityService {
         return quantityAvailableToPromiseByProductNotInBin ?: 0
     }
 
+    Integer getQuantityAvailableToPromiseForProductInBin(InventoryItem inventoryItem, Location location, Location binLocation) {
+        return ProductAvailability.createCriteria().get {
+            projections {
+                property("quantityAvailableToPromise")
+            }
+            eq("inventoryItem", inventoryItem)
+            eq("location", location)
+            if (binLocation) {
+                eq("binLocation", binLocation)
+                return
```

Comment:
Does this return work? :smile:

Reply:
yeah, since we don't have any additional statements below (except `isNull` which we would not want to be executed if bin location is present), this works fine.